### DISCUSSION TITLE
[GSoC] Interest in "8. Add time-series anomaly detection support to Anomalib"

### DISCUSSION THREAD

**Discussion by aadhamm** [openvinotoolkit/openvino#16490] [Google Summer of Code]
Hello Everyone,

my name is Adham Mohammed Nassar, a senior electronics and communication engineering student at Nile University, Egypt. 

Throughout my university journey, I have learned and implemented low-scale projects related to ICT and IoT field (using C++ when it comes to the embedded layer system, Python, Javascript, and sometimes Matlab when it comes to the upper layers and AI applications) and it was always appealing to me how adding AI component in your system can extend its capabilities to aspects the develop the system intelligence and reliability and benefit businesses. I had an experience as a Junior Research Assistant at the Wireless Intelligent Networks Research Center (WINC) at my university.

### Reason for Interest in "8. Add time-series anomaly detection support to Anomalib"
As a matter of fact, my graduation project is **on the same topic** that I want to contribute to your respected program, my Graduation Project is co-operated with one of the large industrial partners in the telecom sector in my country, Orange Egypt, where I was directly communicating with Network Operational Skill Center, which is responsible for validating new technologies and innovative solutions before implementing them on a large scale to the cellular network, we were contributing to the "Self-organizing network" Topic, under a use-case that involved **developing a time-series anomaly detection model** for detecting a specific type of cell degradation named "sleeping cells". [Project Presentation](https://nileuniversity-my.sharepoint.com/:p:/g/personal/al_kahled_nu_edu_eg/EQCJ7CMt0DFHpWystKR7nygBF7n4iNnDroZqDgpFqw7HoA?e=nq2CkZ)

### Some relative experience & achievements
I had some experience in proposing and developing ideas for AI applications in competitions, as I 
- Participated in Vodafone Egypt 1st Ai Assistive Tools Hackathon and got 3rd place award, 
where we Introduced “AccessME”, an AI-assistive indoor navigation mobile application for visually impaired people, it also had diverse and inclusive features that are provided based on the kind of disability and a reporting system based on GPT-3. [Porject Presenation](https://docs.google.com/presentation/d/1mZaTnpcDvpvN5dqyMJlovYtARnChjhSY/edit) - [Project Video](https://drive.google.com/file/d/1KnjRTXnlphg168Vb8ZQdP_UpAbi9JmlX/view)

- Shortlisted to participate in _VOIS International Hackathon 2023, with 30 teams from Egypt, India, & Romania, in which we pitched "Auto-Meter", IoT smart meter solution for Deep Learning based Anomaly Detection, aiming to detect illegal consumption/outliers coming from smart meter uni-variate time-series data, which was mentored by an expert from Valeo company during an IoT academic elective course. [Project Link](https://drive.google.com/file/d/1-wQqsHMLUDoKLRHxffTovHw95hST5dbE/view?usp=sharing)

### Initial plan (proposal)
I have already submitted my [PR prereq task](https://github.com/openvinotoolkit/openvino_notebooks/pull/939) here and I plan to invest my summer in contributing to the chosen topic within your respected program,
- I should propose that I would get to investigate and know Anomalib and its contribution guidelines more
- For the 1st phase, I will begin to implement the required 1-D Pytorch data adapter, by Defining the dataset class, implementing the necessary data preprocessing methods (_len_, _getitem_, setup), defining the PyTorch-Lightning data module, providing dataset batching and shuffling, & maybe finally use the adapter by creating an instance from the PyTorch trainer object.
- for the 2nd phase, I will begin to choose an appropriate algorithm whether it's density-based, reconstruction-based, etc. implement the algorithm and integrate it with Anomalib. Test the implementation on some 1-d datasets, and use appropriate evaluation metrics, such as precision, recall, and a confusion matrix to visualize model performance. Document it and if it's time i may be do a pull request for the respective library.

### I wanted finally to express my gratitude for your invaluable support and guidance as part of the Google Summer of Code (GSoC) 2023 program. 

Thank you.

A.Nassar@nu.edu.eg
[Linkedin](https://www.linkedin.com/in/adham-nassar-032107120/)

**Comment by adrianboguszewski** [comment]
@samet-akcay @djdameln

**Comment by samet-akcay** [comment]
Thanks @aadhamm for you interest! We'll review the proposal

**Reply by aadhamm** [reply]
I didn't submit it yet, I plan to shortly send a first draft to your email to get early feedback and then submit it if that's okay. @samet-akcay
